### PR TITLE
Setup ktlint

### LINKED ISSUE #44: Setup ktlint

### PR DESCRIPTION
**Pull request by antalasztrik**
Closes #44

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -213,6 +213,47 @@
...
+                    <executions>
+                        <execution>
+                            <id>ktlint</id>
+                            <phase>validate</phase>
+                            <configuration>
+                                <target name="ktlint">
+                                    <java taskname="ktlint" dir="${basedir}" fork="true" failonerror="true"
+                                          classpathref="maven.plugin.classpath" classname="com.pinterest.ktlint.Main">
+                                        <arg value="src/**/*.kt"/>
+                                    </java>
```
Comment (battila7) [inline pom.xml:232]: Is this going to lint the test sources as well? It'd be great if we had linting set up for test sources too (even though we don't really have them yet 😄)

File: pom.xml [hunk 2]
```diff
@@ -213,6 +213,47 @@
                     <artifactId>git-commit-id-plugin</artifactId>
                     <version>${git-commit-id-plugin.version}</version>
                 </plugin>
+
+                <plugin>
+                    <groupId>org.apache.maven.plugins</groupId>
+                    <artifactId>maven-antrun-plugin</artifactId>
+                    <version>1.8</version>
```
Comment (battila7) [inline pom.xml]: Can you please extract this version into a property?

File: pom.xml [hunk 3]
```diff
@@ -213,6 +213,47 @@
...
+                                </target>
+                            </configuration>
+                            <goals><goal>run</goal></goals>
+                        </execution>
+                    </executions>
+                    <dependencies>
+                        <dependency>
+                            <groupId>com.pinterest</groupId>
+                            <artifactId>ktlint</artifactId>
+                            <version>0.37.2</version>
```
Comment (battila7) [inline pom.xml]: Can you please extract this version into a property?

**Review by battila7 — CHANGES_REQUESTED** [review]
I absolutely love that linting is up and running now! Great stuff goin' on!

Requested three small changes, fine overall.

File: pom.xml (hunk 1 above)
Comment (antalasztrik) [reply pom.xml:232]: If they will be located at **src/test/kotlin/...** then they are linted as well. Just tested them by adding a test/kotlin directory into the src of a module.

Comment (battila7) [reply pom.xml:232]: Ah, I forgot how Maven projects are structured 😄 (been working with TypeScript lately where tests have their own top-level directory). Nice, thanks!

**Review by battila7 — CHANGES_REQUESTED** [review]
🔥 

(accidentally pressed Request Changes)

**Review by battila7 — APPROVED** [review]

**Pull request merged**